### ISSUE TITLE
Transition to Equinox as a backend

### ISSUE THREAD

**Issue by renecotyfanboy** [renecotyfanboy/jaxspec#73]
haiku won't be improved in the future, but will be maintained to be compatible with newer JAX releases. It would be great to consider switching to equinox as a backend for JAXspec, since it is much lighter and would lead to less spaghetti code in the engine.

**Comment by renecotyfanboy** [comment]
After some tryouts, I figured out a nice code architecture which doesn't rely on weird metaclass as it is done right now. Equinox modules are great and can be instantiated unlike haiku modules. However I can't get it to work right now since I'm not familiar yet with this library. 

UML diagram

![image](https://github.com/renecotyfanboy/jaxspec/assets/49200287/8db021bc-031e-463a-9635-b3136df4346e)

Here, the natural model building SpectralModel = Phabs() * ( Powerlaw() + Blackbody()) is handled by the compositions defined in the Spectral model class. Individual components are built with a single node graph in the ModelComponent class constructor. I'll dive into this later!

**Comment by patrick-kidger** [comment]
I recently came across this library.
I'd love to see Equinox used as the backend :)

**Comment by renecotyfanboy** [comment]
Wow, hi Patrick, I am very honoured to see you there. This project has been in the pipeline for a while, I've been playing around with `equinox` some time ago and it clearly seems feasible to transpose the internal logic of model building based on it. I just need to practise with it a bit more to get used to it, and that should be no problem!

**Comment by renecotyfanboy** [comment]
Hey @patrick-kidger

I am slowly writing the backend transition, and I have a design question. I have many models that requires interpolation on "known in advance" tables. I am passing them as attribute of the eqx modules I am writing and force them to be static and numpy arrays, but in this situation Equinox yell at me saying that JAX array shouldn't be set as static field, while they are 100% numpy arrays. I find it handy to load them as module attributes during instanciation, but don't want gradient computed on them. Is it bad practice to store these tables as attributes for eqx modules ? 

Demo stuff : 

```
class Tbabs(MultiplicativeComponent):
    
    nh = 1.0
    energy :np.array = eqx.field(static=True)
    sigma:np.array = eqx.field(static=True)

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        table = Table.read(table_manager.fetch("xsect_tbabs_wilm.fits"))
        self.energy = np.asarray(table["ENERGY"], dtype=np.float64)
        self.sigma = np.asarray(table["SIGMA"], dtype=np.float64)

    def continuum(self, energy):
        
        sigma = jnp.interp(energy, self.energy, self.sigma, left=1e9, right=0.0)

        return jnp.exp(-self.nh * sigma)
```

**Comment by patrick-kidger** [comment]
JAX treats numpy arrays and JAX arrays pretty much interchangeably, so the warning is correct here -- they shouldn't be static fields either. (This is all a JAX thing, not an Equinox thing, and here Equinox is just trying to prevent you from footgunning yourself.)

If you want these values to be traced arguments to the computation graph then leave them as normal fields. If you want these values to be compiled-in constants then make them global values.

If you don't want gradiens then use `jax.lax.stop_gradient`.
